CrowsZA/functions/fn_spawnIEDClutter.sqf

```diff
@@ -0,0 +1,168 @@
+	iedAmount		- (int) number if explosives to create
+
+Return: none
+
+Spawns item clutter (e.g. garbage piles) and hidden IEDs
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3], ["_radius", 5, [0]], "_density", "_maxClutterSize", "_iedSize", "_iedType", "_iedAmount"];
+
+diag_log format ["CrowsZA-spawnIEDClutter: Spawning a cluster of %1 IEDs at position %2", _iedAmount, _pos];
+
+
+/*/////////////////////////////////////////////////
+ Clutter Types
+/*/////////////////////////////////////////////////
+
+private _smallJunk = [
+	"Land_Garbage_square3_F",
+	"Land_Garbage_square5_F",
+	"Land_Garbage_line_F",
+	"Land_GarbageBarrel_02_buried_F",
+	"Land_Tyre_F"
+];
+
+private _mediumJunk = [
+	"Land_GarbageBags_F",
+	"Land_GarbageBarrel_02_F",
+	"Land_GarbageBarrel_01_english_F",
+	"Land_GarbagePallet_F",
+	"Land_GarbageHeap_04_F",
+	"Land_GarbageHeap_03_F",
+	"Land_GarbageHeap_02_F",
+	"Land_GarbageWashingMachine_F",
+	"Land_Tyres_F"
+];
+
+private _largeJunk = [
+
+];
+
+private _smallElectronics = [
+	"Land_CarBattery_02_F",
+	"Land_Microwave_01_F",
+	"Land_FMradio_F"
+];
+
+private _mediumWrecks = [
+
+];
+
+private _largeWrecks = [
+	"Land_Wreck_BRDM2_F",
+	"Land_Wreck_BMP2_F",
+	"Land_V3S_wreck_F",
+	"Land_Wreck_T72_hull_F",
+	"Land_Wreck_Car2_F",
+	"Land_Wreck_Offroad2_F",
+	"Land_Wreck_UAZ_F"
+];
+
+/*/////////////////////////////////////////////////
+ Compile clutter
+/*/////////////////////////////////////////////////
+
+// TODO: Allow the user to select types of clutter for different settings? (e.g. vegetation? luggage?)
+// private _smallClutter = [];
+// private _mediumClutter = [];
+// private _largeClutter = [];
+
+// if(_junk) then {
+// 	_smallClutter = _smallClutter + _smallJunk;
+// 	_mediumClutter = _mediumClutter + _mediumJunk;
+// 	_largeClutter = _largeClutter + _largeJunk;
+// };
+
+// if(_luggage) then {
+//	 ...etc. etc.
+// };
+
+// For now, just curate it on their behalf
+private _smallClutter = _smallJunk + _smallElectronics;
+private _mediumClutter = _mediumJunk + _mediumWrecks;
+private _largeClutter = _mediumJunk + _largeWrecks;
+
+
+/*/////////////////////////////////////////////////
+ Place clutter
+/*/////////////////////////////////////////////////
+
+// TODO: Amount may require fine-tuning
+private _clutterAmount = 3.14 * _radius * _radius * _density * 0.1;
+for "_i" from 1 to _clutterAmount do {
+	private _safePos = [_pos, 0, _radius, 0.5, 0] call BIS_fnc_findSafePos;
+	
+	if(_safePos isEqualTo [0,0]) then {
+		break;
+	};
+
+	private _clutterSize = random _maxClutterSize;
+	private _clutter = switch (true) do
+	{
+		case (_clutterSize > 0.95): { selectRandom _largeClutter };
+		case (_clutterSize > 0.75): { selectRandom _mediumClutter };
+		default { selectRandom _smallClutter }; 
+	};
+	_clutter = _clutter createVehicle _safePos;
+	_clutter setDir (random 360);
+
+	{
+		_x addCuratorEditableObjects [[_clutter], true];
```

Comment:
``addCuratorEditableObjects`` is a server event and has to be run on the server iirc. ZEN has an event hook for this, so you can use theirs like I do when spawning arsenals:
```sqf
//spawn nato supply create at position
private _crate = "B_supplyCrate_F" createVehicle _posAGL;

//ZEN event that registers the object as editable by zeus. Remember the function wants array not single object
["zen_common_addObjects", [[_crate], objNull]] call CBA_fnc_serverEvent;
```

Wiki shows the function to be server event: 
![image](https://user-images.githubusercontent.com/7889925/193405496-796269bc-523d-45b8-aa0e-b7e2e728b1d0.png)

And you can see ZEN have their eventhandler here, and when you raise it with ``serverEvent`` it only gets run on the server. https://github.com/zen-mod/ZEN/blob/master/addons/common/XEH_postInit.sqf#L358-L368

Reply:
The code snippet I put in with the event is for all curators. The code you had before, is essential what ZEN have running server-side:
```sqf
    [QGVAR(addObjects), {
        params ["_objects", ["_curator", objNull]];

        if (!isNull _curator) exitWith {
            _curator addCuratorEditableObjects [_objects, true];
        };

        {
            _x addCuratorEditableObjects [_objects, true];
        } forEach allCurators;
    }] call CBA_fnc_addEventHandler;
```

As you can see they add it for all curators. So you just need to call that event once and everyone gets it added.